packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_model.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/foundation.dart';
+import 'package:subiquity_client/subiquity_client.dart';
+
+export 'package:subiquity_client/subiquity_client.dart' show ApplicationState;
+
+/// View model for [InstallationSlidesPage].
+class InstallationSlidesModel extends ChangeNotifier {
+  /// Creates an instance with the given client.
+  InstallationSlidesModel(this._client);
+
+  final SubiquityClient _client;
+  var _status = ApplicationStatus();
+
+  /// Whether the installation is complete.
+  bool get isDone => _state == ApplicationState.DONE;
+
+  /// Whether the installation state is unknown.
+  bool get isUnknown => _state == ApplicationState.UNKNOWN;
+
+  /// Whether an error has occurred.
+  bool get hasError => _state == ApplicationState.ERROR;
+
+  /// Whether the installation process is active.
+  bool get isInstalling => !isDone && !hasError;
```

Comment:
There should probably be an additional AND clause to this condition: `_state.index >= ApplicationState.RUNNING` (before that state, it's not really installing).

Reply:
what would be a good description for that state? "Preparing..."?

Reply:
"preparing" sounds reasonable to me.